examples/threading/threading.c

```diff
@@ -27,7 +55,24 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      * return true if successful.
      * 
      * See implementation details in threading.h file comment block
+     *
      */
-    return false;
+
+    struct thread_data* st_thread_data_details = (struct thread_data *)malloc(sizeof(struct thread_data));
+
+    int ret_status = 0;
+    st_thread_data_details->stm_mutex = mutex;
+    st_thread_data_details->stm_wait_to_obtain_ms = wait_to_obtain_ms;
+    st_thread_data_details->stm_wait_to_release_ms = wait_to_release_ms;
+    
+    ret_status = pthread_create(thread,NULL,threadfunc,st_thread_data_details);
+    if(ret_status != 0)
+    {
+	    perror("error");
+	    exit(-1);
```

Comment:
@Rishab-Shah is `exit` the right strategy for error handling here?  Probably overkill to kill the process.

Also who frees `st_thread_data_details` in this code path?